crypto/s2n_cipher.h

```diff
@@ -81,6 +81,7 @@ struct s2n_cipher {
         struct s2n_composite_cipher comp;
     } io;
     uint8_t key_material_size;
+    bool ktls_supported;
```

Comment:
Is there no more work that needs to be done to support a ktls cipher? I thought I remember something about creating a cipher_info struct that needs to be done per-cipher. I could be misunderstanding though. This PR is pretty tiny so there's no harm in adding more if it makes sense.

Reply:
This is only the second task. Its going to get alot more complicated so I would rather keep this PR small. https://github.com/aws/s2n-tls/issues/3711

Also its going to be difficult to test kTLS until its fully functional so this is possibly the last bit of testing for awhile.